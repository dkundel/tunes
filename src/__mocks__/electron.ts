import EventEmitter = require('events');

export class MockIpcRenderer extends EventEmitter {}

export const _ipcRendererEmitter = new MockIpcRenderer();

export const ipcRenderer = {
  send: jest.fn(),
  on: jest.fn((evt: string, callback: Function) =>
    _ipcRendererEmitter.on(evt, callback)
  )
};

export const shell = {
  openExternal: jest.fn()
};
