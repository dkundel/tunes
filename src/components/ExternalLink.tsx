import * as React from 'react';

import { shell } from 'electron';

export default class ExternalLink extends React.Component<any, any> {
  onClick(evt: Event) {
    evt.preventDefault();
    shell.openExternal(this.props.href);
  }

  render() {
    return (
      <a {...this.props} onClick={(evt: Event) => this.onClick(evt)}>
        {this.props.children}
      </a>
    );
  }
}
