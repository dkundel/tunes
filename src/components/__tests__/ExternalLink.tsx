jest.mock('electron');

import * as React from 'react';
import ExternalLink from '../ExternalLink';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import { shell } from 'electron';

describe('test ExternalLink component', () => {
  beforeEach(() => {
    (shell.openExternal as jest.Mock<any>).mockClear();
  });

  test('verify snapshot', () => {
    const component = renderer.create(
      <ExternalLink href="https://foo.bar">Hey</ExternalLink>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('triggers shell', () => {
    const component = shallow(
      <ExternalLink href="https://foo.bar">
        Hey
      </ExternalLink>
    );
    const mockEvent = { preventDefault: jest.fn() };
    component.simulate('click', mockEvent);
    expect(shell.openExternal).toHaveBeenCalledWith('https://foo.bar');
    expect(shell.openExternal).toHaveBeenCalledTimes(1);
    expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
  });
});
