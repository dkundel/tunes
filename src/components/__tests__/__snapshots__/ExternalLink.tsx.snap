// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ExternalLink component verify snapshot 1`] = `
<a
  href="https://foo.bar"
  onClick={[Function]}
>
  Hey
</a>
`;
